### PR TITLE
chore: Enable the mtls integration test

### PR DESCRIPTION
**Pull request by oleorhagen**
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

**Comment by oleorhagen** [comment]
Goes along with: 
* https://github.com/mendersoftware/mtls-ambassador/pull/158
* https://github.com/mendersoftware/deviceauth-enterprise/pull/217

File: tests/tests/test_mtls.py
Comment (lluiscampos) [inline tests/tests/test_mtls.py:127]: Once we are here, what about modernizing the formatting?
```suggestion
            f"p11tool --login --provider=/usr/lib/softhsm/libsofthsm2.so --set-pin={pin} --list-all-privkeys"
```

File: tests/tests/test_mtls.py
Comment (lluiscampos) [inline tests/tests/test_mtls.py:135]: I don't think we need to cleanup anything, really. We don't reuse the device for further tests.

File: tests/tests/test_mtls.py
Comment (lluiscampos) [inline tests/tests/test_mtls.py:188]:
```suggestion
                    logger.info(f"set private key to {key_uri}")
```

**Review by lluiscampos — APPROVED** [review]
Amazing that you got this one under control :rocket:

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:133]: Hmm, yeh. I did this first time around. Agreed. Let's just keep it

File: tests/tests/test_mtls.py
Comment (merlin-northern) [inline tests/tests/test_mtls.py:329]: if we have `get_client_service_name` should we have also something for `mender-authd`?

File: tests/tests/test_mtls.py
Comment (merlin-northern) [inline tests/tests/test_mtls.py:177]: why is that not needed anymore?

File: tests/tests/test_mtls.py
Comment (merlin-northern) [inline tests/tests/test_mtls.py:142]: ok, I do not understand this: if you dont configure the engine, then from what do you load the key?

File: tests/tests/test_mtls.py
Comment (merlin-northern) [inline tests/tests/test_mtls.py:206]: I am having doubts here, isnt it the idea of this test to have the server certificate here? (it has been a long time, and I have not wrote this test in whole, so sorry for stupid quesitons)

File: tests/tests/test_mtls.py
Comment (merlin-northern) [inline tests/tests/test_mtls.py:211]: same as comment below, nit, but maybe it would be nice to have something like `env.device.get_client_service_name` but for mender-authd?

File: tests/tests/test_mtls.py
Comment (merlin-northern) [inline tests/tests/test_mtls.py:106]: we discussed that over slack. all I can add is: the fact that we need to change the test in such fundamental manner means that the code behaves differently. I find it disturbing. what if your secure element does not let you store the public key?

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:329]: We had `get_client_service_name` because at the time, it could be two different ones. Not anymore :)

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:177]: This is exactly why I wanted your eyes on it :)

Why was it needed in the first place?

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:206]: We add the certificate to `ca-certs` now :)

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:106]: Then it won't work with the `v3 provider` we are using atm

File: tests/tests/test_mtls.py
Comment (merlin-northern) [reply tests/tests/test_mtls.py:142]: that cannot be. the engine id is arbitrary and there are many ways to give it in the config, how can it be in the code?

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:106]: This is not a limitation in _our_ code

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:142]: Yes, but there is a default directory, in which it loads modules from

Comment (oleorhagen) [reply tests/tests/test_mtls.py:142]: You can override this with the config if you so desire of course

Comment (oleorhagen) [reply tests/tests/test_mtls.py:142]: Also remember, that we are not testing `Engines` now, but `Providers`

Comment (oleorhagen) [reply tests/tests/test_mtls.py:142]: I used this as my guide:

https://github.com/openssl/openssl/blob/master/README-PROVIDERS.md#loading-providers

And went for the explicit approach. Also please remember that this is not an `Engine` test. They don't exist for us in OpenSSLv3 which is what we are testing here.

Comment (oleorhagen) [reply tests/tests/test_mtls.py:142]: I asked a question on Mender-hub also, to see what our users think about this :)

https://hub.mender.io/t/pkcs11-hsm-support-for-authentication-authorization-and-signature-verification/886/18

Comment (merlin-northern) [reply tests/tests/test_mtls.py:142]: I was not aware that we are deprecating ENgines in the client in this way.

File: tests/tests/test_mtls.py
Comment (merlin-northern) [reply tests/tests/test_mtls.py:106]: this is highly misleading. please change the description of the commit. it is a different test, testing different code, using different API.

**Comment by merlin-northern** [comment]
@oleorhagen I will not block this one, we discussed privately, and also you gave me answers here. you have the approval of Lluis, and with the certificate I would ask Fabio, just to be on the safe side. (he was the  author of the original core test)

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:142]: Well, it's OpenSSL deprecating it. Not us :smile: 

We're just not enabling deprecated API's atm.

**Review by merlin-northern — APPROVED** [review]
please change the commit message, explaining exactly why this test is significantly modified. I am approving as you explained to me that the source of all my doubts lies in the decisions you had made.

File: tests/tests/test_mtls.py
Comment (oleorhagen) [reply tests/tests/test_mtls.py:177]: Alright, did a little bit of digging. The storage-proxy cert is already a part of the demo-cert now.

**Pull request merged**
